server/project/calendars.py

```diff
@@ -0,0 +1,165 @@
+from datetime import datetime as dt, timezone, timedelta
+import datetime
+
+import time
+
+
+class Calendars():
+    """
+    The self.calendar_bit_field  stores the busy times of a calendar, where
+    1 represents a busy time slot and 0 represents free.
+    - Each bit marks a 15 (minimum resolution) min time slot.
+      - The first bit represents 00:00am from Today. The second bit represents
+      - 00:15 from TODAY and so on.
+    """
+
+    DAILY_SLOTS = 96 # 15 min is our minimum resolution so tehre is 96 slots in a day
+    MAX_DAY_MASK = (1 << DAILY_SLOTS) - 1 #bit field, every slot for the day is fliped to 1
+    SLOT_MINUTES = 30 # minutes of slots we want to return
+    MIN_RESOLUTION = 15
+
+    def __init__(self, today=dt.now(), duration=None, next_x_days=60):
```

Comment:
We should consider using `dt.utcnow()` since we should be supplying UTC time to the client or the client should be supplying us with a timezone or UTC offset.